src/homework9/linkedlist/LinkedList.java

```diff
@@ -0,0 +1,185 @@
+package homework9.linkedlist;
+
+public class LinkedList<T> implements List<T> {
+    private Node<T> node = null;
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
+    private static class Node<T> {
+        private Node<T> prev;
+        private Node<T> next;
+        private T element;
+
+        public Node(Node<T> prev, T element, Node<T> next) {
+            this.prev = prev;
+            this.next = next;
+            this.element = element;
+        }
+
+        public Node<T> getPrev() {
+            return prev;
+        }
+
+        public void setPrev(Node<T> prev) {
+            this.prev = prev;
+        }
+
+        public Node<T> getNext() {
+            return next;
+        }
+
+        public void setNext(Node<T> next) {
+            this.next = next;
+        }
+
+        public T getElement() {
+            return element;
+        }
+
+        public void setElement(T element) {
+            this.element = element;
+        }
+    }
+
+    public LinkedList() {
+        size = 0;
+    }
+
+    @Override
+    public void add(T value) {
+        node = new Node<>(node, value, null);
+
+        if (size == 0) {
+            first = node;
+            last = node;
+        } else {
+            last.setNext(node);
+            last = node;
+        }
+        size++;
+    }
+
+    @Override
+    public void add(T value, int index) {
+        if (index > -1 && index <= size) {
+            if (index == size) {
+                add(value);
+            } else {
+                Node<T> necessaryNode = getNodeByIndex(index);
+                node = new Node<>(necessaryNode.getPrev(), value, necessaryNode);
+                necessaryNode.setPrev(node);
+                getNodeByIndex(index - 1).setNext(node);
```

Comment:
better change:
getNodeByIndex(index - 1).setNext(node); -> necessaryNode.getPrev().setNext(node); 
and move it before this line: necessaryNode.setPrev(node);

Reply:
Done.